src/cluster_migrateslots.c

```diff
@@ -737,17 +1020,82 @@ void clusterHandleFlushDuringSlotMigration(void) {
     while ((ln = listNext(&li))) {
         slotMigrationJob *job = (slotMigrationJob *)ln->value;
         if (!isSlotMigrationJobInProgress(job)) continue;
-        /* Note that if we are exporting, we don't send a FAIL message, so the
-         * target should reconnect and complete the migration shortly after,
-         * since we now have no data.
-         *
-         * If we are importing, we fail the migration and expect the operator to
-         * retry. */
-        finishSlotMigrationJob(job, SLOT_MIGRATION_JOB_FAILED,
-                               "Data was flushed");
+        /* Since the data is now empty, the operator should retry and the next
+         * attempt will complete quickly. */
+        finishSlotMigrationJob(job, SLOT_MIGRATION_JOB_FAILED, "Data was flushed");
     }
 }
 
+/* As the final step of a slot import, once all cleanup is done (if needed), we
+ * ensure that replicas track the state transition through propagating SYNCSLOTS
+ * FINISH with the final state.
+ *
+ * Also note, we propagate to AOF as well in this function, allowing us to know
+ * the latest slot import states after restoring from just AOF. */
+void propagateSyncSlotsFinish(slotMigrationJob *job) {
+    /* CLUSTER SYNCSLOTS FINISH STATE <state> NAME <name> (MESSAGE <message>) */
```

Comment:
```suggestion
    /* CLUSTER SYNCSLOTS FINISH STATE <state> NAME <name> [ MESSAGE <message> ] */
```
Historically square brackets are for optional text.

Reply:
Done.